Replace template README with project documentation (#179)
ministryofjustice/hmpps-arns-assessment-platform-api: README.md

# HMPPS ARNS Assessment Platform API

[![repo standards badge](https://img.shields.io/badge/endpoint.svg?&style=flat&logo=github&url=https%3A%2F%2Foperations-engineering-reports.cloud-platform.service.justice.gov.uk%2Fapi%2Fv1%2Fcompliant_public_repositories%2Fhmpps-arns-assessment-platform-api)](https://operations-engineering-reports.cloud-platform.service.justice.gov.uk/public-report/hmpps-arns-assessment-platform-api)
[![Docker Repository on ghcr](https://img.shields.io/badge/ghcr.io-repository-2496ED.svg?logo=docker)](https://ghcr.io/ministryofjustice/hmpps-arns-assessment-platform-api)

> [![API docs](https://img.shields.io/badge/API_docs_-view-85EA2D.svg?logo=swagger)](https://hmpps-arns-assessment-platform-api-dev.hmpps.service.justice.gov.uk/webjars/swagger-ui/index.html?configUrl=/v3/api-docs)

Reviewer: The API docs button doesn't work

Author: Good spot, thanks @alexw-moj , resolved in https://github.com/ministryofjustice/hmpps-arns-assessment-platform-api/pull/179/commits/392001899f30cc55d4c93cf8de3d324248b9baf3

```

The API will be available at http://localhost:8081 with HMPPS Auth on http://localhost:9090

### Running in IntelliJ

Start the supporting services without the API container, then run the Spring Boot application with the `dev` profile:

```bash
docker compose -f docker/docker-compose.base.yml up postgres redis hmpps-auth localstack --wait
```

### Common Development Commands

```bash
# Testing
make test                         # Run all tests
make test-targeted TESTS="*Test"  # Run specific tests

# Code Quality
make lint                         # Run ktlint
make lint-fix                     # Auto-fix linting issues

# Container Management
make down                         # Stop all containers
make clean                        # Clean containers, volumes, and build dirs
make update                       # Update container images

# Database (requires active kubectl context)
make db-port-forward-pod          # Create port-forwarding pod
make db-port-forward              # Forward local port 5434 to remote DB
make db-connect                   # Connect to remote DB via psql

# TypeScript Client
make build-client                 # Generate TypeScript client from OpenAPI spec

> ```

Reviewer 2: The make commands are documented in the Makefile, probs best not to duplicate them here for ease of maintenance 🤔 

```suggestion
View the full list of development commands and their use by running:

```bash
make
```
```

The system stores four core entity types in PostgreSQL:

| Table                    | Purpose                                                                       |
| ------------------------ | ----------------------------------------------------------------------------- |
| `assessment`             | Root entity with UUID and type (e.g. `SENTENCE_PLAN`)                         |
| `assessment_identifier`  | External identifiers linking assessments to upstream systems (e.g. OASys IDs) |
| `event`                  | Immutable append-only event log, JSONB payload with polymorphic `data_type`   |
| `aggregate`              | Versioned snapshots of derived state, rebuilt by replaying events              |
| `timeline`               | User-facing audit trail entries                                               |

### Write Path (Commands)

1. Client sends a `POST /command` with a batch of typed commands
2. The `RetryableCommandDispatcher` dispatches each command through the `CommandBus`
3. The matched `CommandHandler` validates the command, emits one or more `Event`s, and optionally creates timeline entries
4. Events are persisted to the `event` table as immutable JSONB records

> 5. The `AggregateState` replays events through typed `EventHandler`s to rebuild the `AssessmentAggregate`

Reviewer 2: ```suggestion
4. The `AggregateState` replays events through typed `EventHandler`s to rebuild associated aggregates (e.g. `AssessmentAggregate`)
5. Events are persisted to the `event` table as immutable JSONB records
```

> 6. The updated aggregate snapshot is persisted to the `aggregate` table with an incremented version

Reviewer 2: ```suggestion
6. The updated aggregate snapshots are persisted to the `aggregate` table with an incremented version
```
